carpooling/views/views.xml

```diff
@@ -0,0 +1,265 @@
+        </tree>
+      </field>
+    </record>
+
+    <!-- <record id="tree_id" model="ir.actions.act_window.view"> 
+      <field eval="3" name="sequence"/> 
+      <field name="view_mode">tree</field> 
+      <field name="view_id" ref="tree_view_id"/> 
+      <field name="act_window_id" ref="action_id"/> 
+    </record> -->
+
+    <record model="ir.ui.view" id="carpooling.available_trip_form">
+      <field name="name">carpooling form</field>
+      <field name="model">carpooling.vehicle_trip</field>
+      <field name="arch" type="xml">
+        <form string="Trip info" create="false" edit="false" delete="false" duplicate="false">
+          <group colspan="4">
+              <group colspan="2" col="2">
+                  <separator string="Required fields" colspan="2"/>
+                  <field name="driver" readonly="1"/>
+                  <field name="departure_time" required="1"/>
+                  <field name="departure_loc" required="1"/>
+                  <field name="destination_loc" required="1"/>
+                  <field name="available_seats" required="1"/>
+              
+                  <separator string="Optional" colspan="2"/>
+                  <field name="vehicle_type"/>
+                  <field name="description"/>
+
+                  <notebook colspan="4">
+                    <page string="Map">
+                        <field name="description"/>
+                    </page>
+                    <page string="Location pictures">
+                        <field name="description"/>
+                    </page>
+                    <page string="Passengers">
+                        <tree create="false" edit="false" delete="false">
+                          <field name="passengers" readonly="1"/>
+                        </tree>
+                    </page>
+                    <page string="Vehicle pictures">
+                        <field name="description"/>
+                    </page>
+                  </notebook>
+              </group>
+          </group>
+        </form>
+      </field>
+    </record>
+
+    <record model="ir.ui.view" id="carpooling.trip_search">
+      <field name="name">carpooling trip search</field>
+      <field name="model">carpooling.vehicle_trip</field>
+      <field name="arch" type="xml">
+        <search>
+          <field name="departure_time"/>
+          <field name="departure_loc"/>
+          <field name="destination_loc"/>
+          <field name="remaining_seats"/>
+          <field name="passengers"/>
+          <!-- <field name="current_user_is_passenger"/> -->
+          <!-- <field name="current_user_is_passenger" string="Is current user a passenger?" filter_domain="[(current_user_is_passenger,'=', True)]"/> -->
+        </search>
+      </field>
+    </record>
+
+    <!-- Your trips -->
+
+    <record model="ir.ui.view" id="carpooling.your_trips">
```

Comment:
I will not go over all the views, but it would have been more efficient to create only one tree view and one form . 
With default filter, and change it with the different action ( on the menu )